src/ch/ch_conf.c

```diff
@@ -171,6 +171,62 @@ virCHDriverConfigDispose(void *obj)
     g_free(cfg->logDir);
 }
 
+int
+virCHVersionString(const char *str, unsigned long *version,
+                      bool allowMissing)
+{
+    unsigned int major, minor = 0, micro = 0;
+
+    if (strncmp(str, "cloud-hypervisor v", 18) == 0) {
+        /**
+         * Check for regular 'cloud-hypervisor --version' output with major,minor and micro version
+         * This format is for upstream CLH release version
+         * For ex: cloud-hypervisor v32.0.0
+         */
+        if(sscanf(str, "cloud-hypervisor v%d.%d.%d", &major, &minor, &micro) != 3){
+            /**
+             * Try extracting major and minor version only
+             * This format is for upstream CLH main branch build
+             * For ex: cloud-hypervisor v33.0-104-ge0e3779e-dirty
+             */
+            if(sscanf(str, "cloud-hypervisor v%d.%d", &major, &minor) != 2){
+
+                /**
+                 * Log Error if none of the format matches and return -1
+                 */
+                fprintf(stderr, "Can not extract upstream CLH version: %s\n", str);
+                return -1;
+            }
+        }
+    } else {
+        /**
+         * Check for msft 'cloud-hypervisor --version' output
+         * This format is for msft CLH main branch build
+         * For ex: cloud-hypervisor msft/v32.0.131-1-ga5d6db5c-dirty
+         */
+        if(sscanf(str, "cloud-hypervisor msft/v%d.%d.%d", &major, &minor, &micro) != 3){
+            /**
+             * Try extracting major and minor version only for msft CLH
+             * For ex: cloud-hypervisor msft/v33.0-104-ge0e3779e-dirty
+             */
+            if(sscanf(str, "cloud-hypervisor msft/v%d.%d", &major, &minor) != 2){
+                fprintf(stderr, "Can not extract msft CLH version: %s\n", str);
+                return -1;
+            }
+        }
+    }
+
+    // Print the extracted version numbers
+    fprintf(stdout, "Major: %d\n", major);
+    fprintf(stdout, "Minor: %d\n", minor);
+    fprintf(stdout, "Micro: %d\n", micro);
```

Comment:
Use `VIR_INFO` instead, if you want to print this out.

Reply:
Will that print in journalctl log?

Reply:
Addressed in latest push. Please check.